hack/deploy/install.sh

```diff
@@ -132,131 +219,64 @@ echo "PORT=$PORT"
-    for n in "${ns[@]}"; do
-        if [ "$n" = "$NAMESPACE" ]; then
-            export found=1
-        fi
-    done
-    if [ "$found" -eq 0 ]; then
-        kubectl create ns $NAMESPACE
-    fi
-
-    local catalogNames=(${CATALOG_NAMES//[,]/ })
-    for catalog in "${catalogNames[@]}"; do
-        kubectl create configmap $catalog --namespace $NAMESPACE --from-file=hack/deploy/catalogs/$catalog
-    done
-    ${SCRIPT_LOCATION}hack/deploy/deployment.yaml | $ONESSL envsubst | kubectl apply -f -
-    ${SCRIPT_LOCATION}hack/deploy/service.yaml | $ONESSL envsubst | kubectl apply -f -
-    ${SCRIPT_LOCATION}hack/deploy/rbac.yaml | $ONESSL envsubst | kubectl apply -f -
-    ${SCRIPT_LOCATION}hack/deploy/cluster_service_broker.yaml | $ONESSL envsubst | kubectl apply -f -
-
-    echo
-    echo "waiting until service-broker deployment is ready"
-    $ONESSL wait-until-ready deployment $NAME --namespace $NAMESPACE || { echo "service-broker deployment failed to be ready"; exit 1; }
-
-    echo
-    echo "Successfully installed service-broker in $NAMESPACE namespace!"
-}
-
-run() {
-    pushd $REPO_ROOT
-        ensure_onessl
-        deploy_service_broker
-    popd
-}
-
-uninstall() {
-    # delete configmap
-    local catalogNames=(${CATALOG_NAMES//[,]/ })
+if [[ "$UNINSTALL" -eq 1 ]]; then
+     # delete configmap
+    catalogNames=(${CATALOG_NAMES//[,]/ })
     for catalog in "${catalogNames[@]}"; do
-        kubectl delete configmap $catalog --namespace $NAMESPACE
+        kubectl delete configmap ${catalog} --namespace ${NAMESPACE}
     done
     # delete service-broker
-    kubectl delete service -l app=$APP --namespace $NAMESPACE
-    kubectl delete deployment -l app=$APP --namespace $NAMESPACE
+    kubectl delete service -l app=${APP} --namespace ${NAMESPACE}
+    kubectl delete deployment -l app=${APP} --namespace ${NAMESPACE}
     # delete RBAC objects, if --rbac flag was used.
-    kubectl delete serviceaccount -l app=$APP --namespace $NAMESPACE
-    kubectl delete clusterrolebindings -l app=$APP
+    kubectl delete serviceaccount -l app=${APP} --namespace ${NAMESPACE}
+    kubectl delete clusterrolebindings -l app=${APP}
 
     echo
     echo "waiting for service-broker pod to stop running"
     for (( ; ; )); do
-        pods=($(kubectl get pods --all-namespaces -l app=$APP -o jsonpath='{range .items[*]}{.metadata.name} {end}'))
+        pods=($(kubectl get pods --all-namespaces -l app=${APP} -o jsonpath='{range .items[*]}{.metadata.name} {end}'))
         total=${#pods[*]}
-        if [ $total -eq 0 ] ; then
+        if [[ ${total} -eq 0 ]] ; then
             break
         fi
         sleep 2
     done
 
-    kubectl delete clusterservicebroker -l app=$APP
-    kubectl delete ns $NAMESPACE
+    kubectl delete clusterservicebroker -l app=${APP}
+    kubectl delete ns ${NAMESPACE}
 
     echo
     echo "Successfully uninstalled service-broker!"
     exit 0
-}
+fi
+
+found=0
```

Comment:
This assumes users have permission to List & Create namespaces. We have left of those outside the script. Use should create those beforehand.